src/main.cpp

```diff
@@ -2825,7 +2837,7 @@ void static UpdateTip(CBlockIndex *pindexNew) {
         }
         else {
             // Auto Backup defined so check block height
-            if (chainActive.Height() >= BackupBlock )
+            if (chainActive.Height() == BackupBlock )
```

Comment:
Changing this to a strict equality prevents a backup file being written at a later-than specified block.
If we are past the designated backup block, we should simply assume it is unsafe to back up the wallet.